### PR TITLE
[WFLY-12485] Galleon layers for embedded infinispan-based web cluster…

### PR DESCRIPTION
**Pull request by bstansberry**
…ing and JPA with a distributed 2nd level cache

https://issues.jboss.org/browse/WFLY-12485

### CODE REVIEW

**Comment by bstansberry** [comment]
@jfdenise Please review.

**Comment by bstansberry** [comment]
And @pferraro :)

As I mentioned in a chat message, a 'jgroups' layer may make sense.

**Comment by bstansberry** [comment]
I just realized I didn't update the doc page so I killed the CI jobs and will resubmit.

File: testsuite/layers/pom.xml [hunk 1]
```diff
@@ -1058,6 +1058,58 @@
...
                         </configuration>
+                    </execution>
+                    <execution>
+                        <id>jpa-distributed-provisioning</id>
+                        <goals>
+                            <goal>provision</goal>
+                        </goals>
+                        <phase>compile</phase>
+                        <configuration>
+                            <install-dir>${layers.install.dir}/jpa</install-dir>
```
Comment (jfdenise) [inline testsuite/layers/pom.xml]: Should be jpa-distributable

File: testsuite/layers/pom.xml [hunk 2]
```diff
@@ -1319,6 +1371,58 @@
...
                         </configuration>
                     </execution>
+                    <execution>
+                        <id>web-clustering-provisioning</id>
+                        <goals>
+                            <goal>provision</goal>
+                        </goals>
+                        <phase>compile</phase>
+                        <configuration>
+                            <install-dir>${layers.install.dir}/jpa</install-dir>
```
Comment (jfdenise) [inline testsuite/layers/pom.xml]: should be web-clustering

Comment (jfdenise) [reply testsuite/layers/pom.xml]: You should add the 2 new layers to all-layers-provisioning-full execution (end of this file).

**Comment by bstansberry** [comment]
I added the doc page content.

@pferraro @jfdenise One thing to think about is what we would call a layer that used hotrod based web session caching. The impl details of this PR are quite simple; I think the main thing to think about is stuff like names to ensure we don't paint ourselves into a corner.

File: galleon-pack/src/main/resources/feature_groups/distributable-web.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+<feature-group-spec name="distributable-web" xmlns="urn:jboss:galleon:feature-group:1.0">
+
+    <feature spec="subsystem.distributable-web">
+        <param name="default-session-management" value="default"/>
+        <param name="default-single-sign-on-management" value="default"/>
+        <feature spec="subsystem.distributable-web.infinispan-session-management">
+            <param name="infinispan-session-management" value="default"/>
+            <param name="cache-container" value="web"/>
+            <param name="granularity" value="SESSION"/>
+            <feature spec="subsystem.distributable-web.infinispan-session-management.affinity.primary-owner"/>
```
Comment (pferraro) [inline galleon-pack/src/main/resources/feature_groups/distributable-web.xml]: We probably want a separate non-clustered version of this feature-group-spec configured to use spec="subsystem.distributable-web.infinispan-session-management.affinity.local".
While "primary-owner" will fallback to equivalent behavior, it is likely going to be confusing.

File: galleon-pack/src/main/resources/feature_groups/distributable-web.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        </feature>
+        <feature spec="subsystem.distributable-web.infinispan-single-sign-on-management">
+            <param name="infinispan-single-sign-on-management" value="default"/>
+            <param name="cache-container" value="web"/>
+            <param name="cache" value="sso"/>
+        </feature>
+        <feature spec="subsystem.distributable-web.routing.infinispan">
```
Comment (pferraro) [inline galleon-pack/src/main/resources/feature_groups/distributable-web.xml]: A non-clustered version of this feature-spec-group would want to use:
<feature spec="subsystem.distributable-web.routing.local"/>

File: galleon-pack/src/main/resources/feature_groups/distributable-web.xml (hunk 3 above)
Comment (bstansberry) [reply galleon-pack/src/main/resources/feature_groups/distributable-web.xml]: @pferraro What would that feature-group be called?

Comment (pferraro) [reply galleon-pack/src/main/resources/feature_groups/distributable-web.xml]: distributable-web-local?

**Comment by bstansberry** [comment]
I'm going to close this for now as this is WF 19 stuff and #12599 has a test-only variant of this for use in WF !8 tests. So keep it simple and deal with 18 work first.

**Pull request closed without merging**